docs/namers.md

```diff
@@ -0,0 +1,136 @@
+| 8 | File Manager | file, files, ftp, bmp, windows, program, directory, bitmap, win3, zip |
+| 9 | Printer and Fonts | printer, print, fonts, printing, font, printers, hp, driver, deskjet, prints |
+
+### Using OpenAI's LLMs
+
+You might not have the computational resources to run a high-quality LLM locally.
+Luckily Turftopic allows you to use OpenAI's chat models for topic naming too!
+
+
+!!! info
+    You will also need to install the `openai` Python package.
+    ```bash
+    pip install openai
+    export OPENAI_API_KEY="sk-<your key goes here>"
+    ```
+
+```python
+from turftopic.namers import OpenAITopicNamer
+
+namer = OpenAITopicNamer("gpt-4o-mini")
+model.rename_topics(namer)
+model.print_topics()
+```
+
+| Topic ID | Topic Name | Highest Ranking |
+| - | - | - |
+| 0 | Operating Systems and Software  | windows, dos, os, ms, microsoft, unix, nt, memory, program, apps |
+| 1 | Atheism and Belief Systems | atheism, atheist, atheists, belief, religion, religious, theists, beliefs, believe, faith |
+| 2 | Computer Architecture and Performance | motherboard, ram, memory, cpu, bios, isa, speed, 486, bus, performance |
+| 3 | Storage Technologies | disk, drive, scsi, drives, disks, floppy, ide, dos, controller, boot |
+| 4 | Moral Philosophy and Ethics | morality, moral, objective, immoral, morals, subjective, morally, society, animals, species |
+| 5 | Christian Faith and Beliefs | christian, bible, christians, god, christianity, religion, jesus, faith, religious, biblical |
+| 6 | Serial Modem Connectivity | modem, port, serial, modems, ports, uart, pc, connect, fax, 9600 |
+| 7 | Graphics Card Drivers | card, drivers, monitor, vga, driver, cards, ati, graphics, diamond, monitors |
+| 8 | Windows File Management | file, files, ftp, bmp, windows, program, directory, bitmap, win3, zip |
+| 9 | Printer Font Management | printer, print, fonts, printing, font, printers, hp, driver, deskjet, prints |
+
+### Prompting
+
+Since these namers use chat-finetuned LLMs you can freely define custom prompts for topic name generation:
+
+```python
+from turftopic.namers import OpenAITopicNamer
+
+system_prompt = """
+You are a topic namer. When the user gives you a set of keywords, you respond with a name for the topic they describe.
+You only repond briefly with the name of the topic, and nothing else.
+"""
+
+prompt_template = """
+You will be tasked with naming a topic.
+Based on the keywords, create a short label that best summarizes the topics.
+Only respond with a short, human readable topic name and nothing else.
+
+The topic is described by the following set of keywords: {keywords}.
+"""
+
+namer = OpenAITopicNamer("gpt-4o-mini", prompt_template=prompt_template, system_prompt=system_prompt)
+```
+
+## N-gram Patterns
+
+You can also name topics based on the semantically closest n-grams from the corpus to the topic descriptions.
+This method typically results in lower quality names, but might be good enough for your use case.
```

Comment:
This looks like a nice default to add. How time-consuming is it?

Reply:
I wouldn't add it as a default honestly. For one I think it's better if topic naming is opt-in, rather than opt-out, and also because these descriptions are not of the highest quality, and they require you to learn and embed a lot of n-grams in a corpus, which might slow you down significantly, especially with fast models like $S^3$